crud/src/main/java/com/redhat/lightblue/mediator/Mediator.java

```diff
@@ -115,7 +114,7 @@ public Mediator(Metadata md,
      * implementation can perform further validations.
      */
     @StopWatch(loggerName = "stopwatch.com.redhat.lightblue.mediator.Mediator", sizeCalculatorClass = "com.redhat.lightblue.mediator.ResponsePayloadSizeCalculator")
-    public Response insert(InsertionRequest req) {
+    public Response insert(InsertionRequest req, RequestMetrics.Context metricCtx) {
```

Comment:
So we decided to track requests at REST layer and parse mediator exceptions via response objects itself. This approach will remove passing or setting of this metricCtx object altogether.